### PR TITLE
feat: add migration to ve

### PR DESCRIPTION
**Pull request by pandadefi**
## Description

Add a function to migrate funds to a new ve contract.
The receiving contract will have to take care of creating a new lock, it can fetch lock information on the existing ve contract.

Fixes # (issue)

## Checklist

- [ ] I have run vyper and solidity linting
- [ ] I have run the tests on my machine
- [ ] I have followed commitlint guidelines
- [ ] I have rebased my changes to the latest version of the main branch

### CODE REVIEW

File: contracts/VotingEscrow.vy [hunk 1]
```diff
@@ -50,6 +50,8 @@ interface IVeYfiRewards:
     def rewardCheckpoint(_account: address) -> bool: nonpayable
     def queueNewRewards(_amount: uint256) -> bool: nonpayable
 
+interface Migrator:
```
Comment (rareweasel) [inline contracts/VotingEscrow.vy:53]: Question:

Should we have a `IMigrator.sol`?

**Review by rareweasel — CHANGES_REQUESTED** [review]
Great job @pandadefi

File: contracts/VotingEscrow.vy [hunk 2]
```diff
@@ -597,6 +611,29 @@ def force_withdraw():
...
+@nonreentrant('lock')
+def migrate():
+    assert(self.next_ve_contract != ZERO_ADDRESS) # dev: no next ve contract
+    assert(self.unlocked == False) # dev: funds are unlocked
+
+    _locked: LockedBalance = self.locked[msg.sender]
+    assert block.timestamp < _locked.end, "lock expired"
+    value: uint256 = convert(_locked.amount, uint256)
+
+    ERC20(self.token).approve(self.next_ve_contract, value)
```
[the remark this answers is not in the document]
Comment (pandadefi) [reply contracts/VotingEscrow.vy:624]: We are making sure the lock still exists [here](https://github.com/yearn/veYFI/pull/100/files#diff-244a97f574a321ee15a9bbcde95f07842637249f93f58dec8c39f6587efd272dR621)
You can't create a lock of zero value [here](https://github.com/yearn/veYFI/blob/master/contracts/VotingEscrow.vy#L439)

File: contracts/VotingEscrow.vy (hunk 1 above)
Comment (pandadefi) [reply contracts/VotingEscrow.vy:53]: We will add it when we need to migrate.

**Pull request merged**